src/stores/analyticsContext.tsx

```diff
@@ -111,6 +119,34 @@ export const AnalyticsProvider = ({
     push(pushParams)
   }
 
+  // Track base location
+  const trackBaseLocation = (baseLocation: string): void => {
+    if (debug === true) {
+      console.debug(`ANALYTICS(trackBaseLocation):`, baseLocation)
+    }
+
+    // Track base location of user:
+    // setCustomDimension is required
+    // the 1 here corresponds to the id of the created custom dimension in the Matomo dashboard
+    // baseLocation comes from the personnel data api
+    push(['setCustomDimension', 1, baseLocation])
+    push(['trackPageView'])
+  }
+
+  // Track rank
+  const trackRank = (rank: string): void => {
+    if (debug === true) {
+      console.debug(`ANALYTICS(trackRank):`, rank)
+    }
+
+    // Track rank of user:
+    // setCustomDimension is required
+    // the 2 here corresponds to the id of the created custom dimension in the Matomo dashboard
+    // rank comes from the personnel data api
+    push(['setCustomDimension', 2, rank])
+    push(['trackPageView'])
```

Comment:
Same question as on the other PR https://github.com/USSF-ORBIT/ussf-portal-client/pull/1100#discussion_r1337879155